src/config.rs

```diff
@@ -292,18 +277,35 @@ impl GitHub {
             }
             matches.push(Match {
                 re,
+                cmd,
                 email,
                 queuekind,
                 secret,
                 timeout,
             });
         }
 
-        if let Some(reposdir) = reposdir {
-            Ok(GitHub { reposdir, matches })
-        } else {
-            Err("A directory for per-repo programs must be specified".to_owned())
+        Ok(GitHub { matches })
+    }
+
+    /// Verify that the `cmd` string is valid, returning `Ok())` if so or `Err(String)` if not.
+    pub(crate) fn verify_cmd_str(cmd: &str) -> Result<(), String> {
+        let mut i = 0;
+        while i < cmd.len() {
+            if cmd[i..].starts_with("%") {
+                if i + 1 == cmd.len() {
+                    return Err("'cmd' cannot end with a single '%'.".to_owned());
+                }
+                let c = cmd[i + 1..].chars().nth(0).unwrap();
+                if !(c == 'e' || c == 'o' || c == 'r' || c == 'j' || c == '%') {
```

Comment:
What if you need a literal `%e` (for example) in the command which you don't want interpreted by snare?

Reply:
`%%e`

Reply:
`date +%e` would make a nice test case.

Reply:
`date +%%e`

Reply:
Yep, agreed. Missed this in the manual first time around.